libsolidity/inlineasm/AsmParser.cpp

```diff
@@ -227,6 +229,18 @@ assembly::VariableDeclaration Parser::parseVariableDeclaration()
 	return varDecl;
 }
 
+assembly::ForLoop Parser:parseForLoop()
+{
+	ForLoop forLoop = createWithLocation<ForLoop>();
+	expectToken(Token::For);
+	forLoop.preCondition = parseBlock();
+	forLoop.expression = parseExpression();
+	forLoop.postCondition = parseBlock();
```

Comment:
At some point it was considered to have this either a block or a statement, but I think it is better to always force a block, also to increase readability ,right?